### DISCUSSION TITLE
Scope of basicManipulationIpc.insertGeometricElement

### DISCUSSION THREAD

**Discussion by Ellord207** [iTwin/itwinjs-core#6682] [Q&A]
We are currently creating an iModel editing app.  During our workflows, users will need to insert geometric and not geometric elements into the iModel.  We have observed the basicManipulationIpc provided by the itwin/editor packages and it's insertGeometricElement method.  That method in particular has no special logic for handling geometric elements.

Would you recommend limiting our use of that method to just geometric elements, and creating our own interface for other elements, or just using that method although the name implies otherwise?  Also, if we are recommended to use it generally, is there a plan to rename it as the methods are still in beta?

**Comment by bbastings** [comment]
You should make your own EditCommandIpc specific to your requirements.
